components/Auth/AuthProvider.js

```diff
@@ -11,24 +13,43 @@ export function AuthProvider({ children }) {
   const [isLoading, setLoading] = useState(false);
   const [isFirstLoading, setFirstLoading] = useState(true);
 
-  useEffect(() => {
+  function setToken(token) {
+    localStorage.clear();
+    localStorage.setItem("token", token);
+    api.API.defaults.headers.common["Authorization"] = `Bearer ${token}`;
```

Comment:
Since in line 5 you import API, you can use it here.

```suggestion
    API.defaults.headers.common["Authorization"] = `Bearer ${token}`;
```